constellation-app/constellation
# Modify storage of Layers view queries and preferences

### Description of the Change
This change fixes #589 and implements a future on saving of queries to ensure only one selection is needed to have a layer become activated.

LayerQueries list which used to be stored inside of StoreGraph was causing issues as the DualGraph managed two instances of StoreGraphs which were out of sync - causing odd behaviour like changing of attribute values. 
The fix stores the list of queries inside a GraphAttribute Object. The same migration process was done to LayerPreferences to fix an issue which arose from the above implementation. 


### Alternate Designs
Trigger a commit whenever one StoreGraph layerQueries list was updated, to force the other StoreGraph within the DualGraph to sync with the updated one. I did not get far into this idea before running into problems.


### Why Should This Be In Core?
Bug Fix - enhances the usability of layers view


### Benefits
layers now work on first click, and react normally because the two graphs are in sync


### Possible Drawbacks
None realised


### Verification Process
Testing Queries
Add color and other attribute queries to each layer, verify correct results
2.	selecting a section of the graph, right click context menu -> add to a vacant layer
a.	Ensure the selection is correct with what is added to the layer
3.	Select more elements and test remove functionality
a.	Ensure the elements are removed from that layer
4.	Test layers with short or long names for visuals and any errors.
Testing Draw mode
1.	Create a new graph
a.	Add a few elements and transactions on
i.	Should appear visible
b.	Open layers view
i.	Select layer 2 only
ii.	Add elements (Vertices and Transactions)
iii.	Elements should only appear on layer 2


### Applicable Issues
#589
